pkg/ebpf/generic_map.go

```diff
@@ -0,0 +1,352 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package ebpf
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+	"unsafe"
+
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const defaultBatchSize = 100
+
+var (
+	batchAPISupportedOnce sync.Once
+	batchAPISupported     bool
+)
+
+// BatchAPISupported returns true if the kernel supports the batch API for maps
+func BatchAPISupported() bool {
```

Comment:
There is a nice wrapper for functions like this in `pkg/util/funcs.MemoizeNoError`, which changes this to:
```
var BatchAPISupported = funcs.MemoizeNoError(func() bool {
...
})